src/hiccup/compiler.clj

```diff
@@ -73,37 +73,48 @@
   (or content
       (and (html-mode?) (not (void-tags tag)))))
 
-(defn- merge-classes [class classes]
+(defn- merge-classes [class classes compiling?]
```

Comment:
Ideally we don't want to check a boolean here, but to run a different function. So for example:

```clojure
(defn merge-classes [class classes]
  (cond
    (nil? class)    classes
    (string? class) (str classes " " class)
    :else           (str classes " " (str/join " " class))))

(defn- merge-classes-form [class classes-form]
  (if (literal? classes-form)
    (merge-classes class classes-form)
    `(merge-classes ~class ~classes-form)))
```